src/Umbraco.Core/Configuration/GlobalSettings.cs

```diff
@@ -397,21 +400,34 @@ public bool UseHttps
         /// An int value representing the time in milliseconds to lock the database for a write operation
         /// </summary>
         /// <remarks>
-        /// The default value is 1800 milliseconds
+        /// The default value is 5000 milliseconds
         /// </remarks>
         /// <value>The timeout in milliseconds.</value>
         public int SqlWriteLockTimeOut
         {
             get
             {
-                try
-                {
-                    return int.Parse(ConfigurationManager.AppSettings[Constants.AppSettings.SqlWriteLockTimeOut]);
-                }
-                catch
+                if (_sqlWriteLockTimeOut != default) return _sqlWriteLockTimeOut;
+
+                var timeOut = 5000; // 5 seconds
+                var appSettingSqlWriteLockTimeOut = ConfigurationManager.AppSettings[Constants.AppSettings.SqlWriteLockTimeOut];
+                if(int.TryParse(appSettingSqlWriteLockTimeOut, out var configuredTimeOut))
                 {
-                    return 1800;
+                    // Only apply this setting if it's not excessively high or low
+                    const int minimumTimeOut = 100;
+                    const int maximumTimeOut = 20000;
+                    if (configuredTimeOut >= minimumTimeOut && configuredTimeOut <= maximumTimeOut) // between 0.1 and 20 seconds
+                    {
+                        timeOut = configuredTimeOut;
+                    }
+                    else
+                    {
+                      Current.Logger.Warn<GlobalSettings>($"The `Umbraco.Core.SqlWriteLockTimeOut` setting in web.config is not between the minimum of {minimumTimeOut} ms and maximum of {maximumTimeOut} ms, defaulting back to {timeOut}");
+                    }
                 }
```

Comment:
@nul800sebastiaan should it use `Constants.AppSettings.SqlWriteLockTimeOut` here instead?

Reply:
Nice one! Updated. 👍